src/gpu/nvidia/cudnn_matmul_executor.hpp

```diff
@@ -367,7 +470,224 @@ struct cudnn_matmul_exec_t : public cudnn_matmul_exec_base_t {
+                    ::sycl::access::mode::read_write>();
+
+            auto arg_src_scale
+                    = CTX_IN_SYCL_MEMORY(DNNL_ARG_ATTR_SCALES | DNNL_ARG_SRC);
+            auto arg_wei_scale = CTX_IN_SYCL_MEMORY(
+                    DNNL_ARG_ATTR_SCALES | DNNL_ARG_WEIGHTS);
+            auto arg_dst_scale
+                    = CTX_IN_SYCL_MEMORY(DNNL_ARG_ATTR_SCALES | DNNL_ARG_DST);
+
+            interop_task(matmul_impl_, engine, cgh, cuda_stream, arg_wt,
+                    arg_src, arg_dst, arg_bias, arg_algo_scratch,
+                    arg_bias_scratch, arg_block_a_scratch, arg_block_b_scratch,
+                    arg_block_c_scratch, arg_src_scale, arg_wei_scale,
+                    arg_dst_scale);
+        });
+    }
+};
+struct cudnn_matmul_lt_runtime_args_algo_scratch_exec_t
+    : public cudnn_matmul_scratch_runtime_args_base_exec_t {
+    status_t execute(const exec_ctx_t &ctx, impl::engine_t *engine,
+            const std::shared_ptr<cudnn_matmul_base_impl_t> matmul_impl_,
+            std::size_t algo_scratch_size,
+            std::size_t bias_scratch_size) override {
+
+        nvidia::stream_t *cuda_stream
+                = utils::downcast<nvidia::stream_t *>(ctx.stream());
+
+        init_scratch_buffers(bias_scratch_size, algo_scratch_size);
+
+        return cuda_stream->interop_task([=](::sycl::handler &cgh) {
+            auto arg_src = CTX_IN_SYCL_MEMORY(DNNL_ARG_SRC);
+            auto arg_wt = CTX_IN_SYCL_MEMORY(DNNL_ARG_WEIGHTS);
+            auto arg_dst = CTX_OUT_SYCL_MEMORY(DNNL_ARG_DST);
+            auto arg_bias = CTX_IN_SYCL_MEMORY(DNNL_ARG_BIAS);
+            auto arg_algo_scratch = xpu::sycl::interop_memory_arg_t<
+                    ::sycl::access::mode::read_write>(*algo_scratch_buff_, cgh);
+            auto arg_bias_scratch = xpu::sycl::interop_memory_arg_t<
+                    ::sycl::access::mode::read_write>();
+            auto arg_block_a_scratch = xpu::sycl::interop_memory_arg_t<
+                    ::sycl::access::mode::read_write>();
+            auto arg_block_b_scratch = xpu::sycl::interop_memory_arg_t<
+                    ::sycl::access::mode::read_write>();
+            auto arg_block_c_scratch = xpu::sycl::interop_memory_arg_t<
+                    ::sycl::access::mode::read_write>();
+
+            auto arg_src_scale
+                    = CTX_IN_SYCL_MEMORY(DNNL_ARG_ATTR_SCALES | DNNL_ARG_SRC);
+            auto arg_wei_scale = CTX_IN_SYCL_MEMORY(
+                    DNNL_ARG_ATTR_SCALES | DNNL_ARG_WEIGHTS);
+            auto arg_dst_scale
+                    = CTX_IN_SYCL_MEMORY(DNNL_ARG_ATTR_SCALES | DNNL_ARG_DST);
+
+            interop_task(matmul_impl_, engine, cgh, cuda_stream, arg_wt,
+                    arg_src, arg_dst, arg_bias, arg_algo_scratch,
+                    arg_bias_scratch, arg_block_a_scratch, arg_block_b_scratch,
+                    arg_block_c_scratch, arg_src_scale, arg_wei_scale,
                     arg_dst_scale);
```

Comment:
IIUC, all these execute functions are almost identical. Would it make sense to use a single execute function in base class in order to avoid boilerplate?
This common execute function could use conditionals to guard some pieces of code, for example:
```c++
if (has_runtime_dims())
    init_scratch_buffers(bias_scratch_size, algo_scratch_size);
```

Reply:
[Commit](https://github.com/oneapi-src/oneDNN/pull/1972/commits/216a0d3de90e2222b64f93e2600dba3e283082f8) which addresses executors and removes dupication